Extension/src/LanguageServer/copilotProviders.ts

```diff
@@ -34,35 +38,109 @@ export async function registerRelatedFilesProvider(): Promise<void> {
+                                    if (filterMap !== undefined) {
+                                        const directAskMap: { [key: string]: string } = JSON.parse(context.flags.copilotcppCompilerArgumentDirectAskMap as string ?? '{}');
+                                        let directAsks: string = '';
+                                        const remainingArguments: string[] = [];
+
+                                        for (const key in filterMap) {
+                                            if (!key) {
+                                                continue;
+                                            }
+                                            const filter = new RegExp(key);
+                                            const matchedArguments = cppContext.compilerArguments.filter(arg => filter?.test(arg));
+                                            if (matchedArguments.length > 0) {
+                                                // Use the last one in case of multiple match.
+                                                if (directAskMap[matchedArguments[matchedArguments.length - 1]]) {
+                                                    directAsks += `${directAskMap[matchedArguments[matchedArguments.length - 1]]} `;
+                                                } else {
+                                                    remainingArguments.push(matchedArguments[matchedArguments.length - 1]);
+                                                }
+                                            } else if (filterMap[key]) {
+                                                // Use the prompt text in the absence of argument.
+                                                directAsks += `${filterMap[key]} `;
+                                            }
+                                        }
+
+                                        const compilerArgumentsValue = remainingArguments.join(", ");
+                                        traits.push({ name: "compilerArguments", value: compilerArgumentsValue, includeInPrompt: true, promptTextOverride: `The compiler arguments include: ${compilerArgumentsValue}.` });
+
+                                        if (directAsks) {
+                                            traits.push({ name: "directAsks", value: directAsks, includeInPrompt: true, promptTextOverride: directAsks });
+                                        }
+                                    }
+                                }
+
+                                traits.push({ name: "intelliSenseDisclaimerEnd", value: '', includeInPrompt: true, promptTextOverride: `End of IntelliSense information.` });
+
+                                const includeTraitsArray = context.flags.copilotcppIncludeTraits as string[] ?? [];
+                                const includeTraits = new Set(includeTraitsArray);
+                                telemetryProperties["includeTraits"] = includeTraitsArray.join(' ');
```

Comment:
Do any of the traits have spaces in them? When logging telemetry you want to make sure you have a stable delimiter.

Reply:
Use comma as delimiter, thanks.